webxr/glwindow/mod.rs

```diff
@@ -125,6 +125,14 @@ impl Device for GlWindowDevice {
         size: UntypedSize2D<i32>,
         sync: Option<GLsync>,
     ) {
+        // Perform the sync while the old window's GL context is active,
+        // or it won't actually do anything - the XR window's GL context
+        // doesn't recognize the sync object and triggers GL_INVALID_OPERATION.
+        if let Some(sync) = sync {
+            self.gl.wait_sync(sync, 0, gl::TIMEOUT_IGNORED);
+            debug_assert_eq!(self.gl.get_error(), gl::NO_ERROR);
+        }
```

Comment:
Does this assume that the main browser window is active when `render_animation_frame` is called?

Reply:
Presumably, yes.